controller/state.go

```diff
@@ -294,14 +294,17 @@ func (m *appStateManager) getRepo(repoURL string) *v1alpha1.Repository {
 	return repo
 }
 
-func (m *appStateManager) persistRevisionHistory(app *v1alpha1.Application, revision string, overrides *[]v1alpha1.ComponentParameter) error {
+func (m *appStateManager) persistRevisionHistory(app *v1alpha1.Application, revision string, overrides []v1alpha1.ComponentParameter) error {
 
 	var nextID int64 = 0
 	if len(app.Status.History) > 0 {
 		nextID = app.Status.History[len(app.Status.History)-1].ID + 1
 	}
+	if overrides == nil {
```

Comment:
if overrides isn't a pointer anymore, how can that variable be nil

Reply:
Oh it's an array 🤦‍♂️

Reply:
Yeah, https://github.com/argoproj/argo-cd/pull/979/files#diff-834370de48b922435d85e0231a6b8499R294 shows how it could happen